js/private/js_run_binary.bzl

```diff
@@ -176,16 +238,31 @@ def js_run_binary(
     if data != None:
         fail("Use srcs instead of data in js_run_binary: https://github.com/aspect-build/rules_js/blob/main/docs/js_run_binary.md#js_run_binary-srcs.")
 
-    # Copy srcs to bin
     extra_srcs = []
+
+    # Hoist js provider files to DefaultInfo
+    hoist = include_transitive_sources or include_transitive_declarations or include_npm_deps
+    if hoist:
+        hoist_to_default_info_name = "{}_hoist_srcs_to_default_info".format(name)
+        _hoist_to_default_info(
+            name = hoist_to_default_info_name,
+            srcs = srcs,
+            include_transitive_sources = include_transitive_sources,
+            include_transitive_declarations = include_transitive_declarations,
+            include_npm_deps = include_npm_deps,
+            tags = kwargs.get("tags"),
+        )
```

Comment:
🤔 `testonly`?

Reply:
I created a new js_info_files rule that does this now